add documentation for switching color themes

toalba opened the pull request.

toalba committed:
add documentation for switching color themes with Plone 6 and Bootstrap 5.3

lenadax committed:
extend color mode documentation

stevepiercy committed:
Tidy up color-mode.md
Also hit up static-resources.md for style

stevepiercy committed:
Merge branch '6-dev' into color-mode-bootstrap5

stevepiercy reviewed (dismissed):
I approve, but I would like @MrTango or another content expert from the Classic UI team to review.

If we don't hear back within a week, please ping me to remind me, and I'll go ahead and merge this. Thank you!

petschki reviewed (changes requested):
Documentation is good, but it's based on the not yet released barceloneta/mockup/staticresources versions introducing Bootstrap 5.3 ... we agreed at the alpine city sprint to release this as minor feature versions (semantic) and ship it with Plone 6.1 but you can use those versions also in Plone 6.0 ... @toalba it would be good to mention this in the docs somewhere.

MrTango reviewed (dismissed):
what petschki says ;)

MrTango committed:
Merge branch '6-dev' into color-mode-bootstrap5

toalba committed:
Update the docs for 6.1 and description for 6.0

Now the docs is for 6.1 and the including of the barceloneta=3.1.0`and plone.staticresources=2.1.0 releases are mentioned.

stevepiercy commented on docs/classic-ui/theming/color-mode.md:
```diff
@@ -141,8 +141,12 @@ See the following example.
   <div class="mb-3">
     <label for="exampleInputEmail1" class="form-label">Email address</label>
-    <input type="email" class="form-control" id="exampleInputEmail1" aria-describedby="emailHelp">
+    <input type="email" class="form-control" id="exampldescribtioneInputEmail1" aria-describedby="emailHelp">
```
What happened here? Did you mean `exampleDescriptionInputEmail1`?

toalba replied:
This is just a example on using the `data-bs-theme` attribute on a from. I don't know if the difference of `id="exampleInputEmail1"` and `id="exampldescribtioneInputEmail1"` is important, but i am new to docs so if you say, i add it.

stevepiercy replied:
`exampldescribtioneInputEmail1` looks like a typographical mistake to me. That's why I asked. I think the original `exampleInputEmail1` is descriptive enough.

stevepiercy commented on docs/classic-ui/theming/color-mode.md:
```diff
@@ -12,16 +12,16 @@ myst:
 
 Bootstrap 5.3 has introduced [Color Modes](https://getbootstrap.com/docs/5.3/customize/color-modes/).
-This chapter is a guide for how to implement color themes in Plone 6.
+This chapter is a guide for how to implement color themes in Plone 6.1.
```
````suggestion
This chapter is a guide for how to implement color modes in Plone 6.1.

```{versionadded} Plone 6.1
```

````

stevepiercy commented on docs/classic-ui/theming/color-mode.md:
```diff
@@ -12,16 +12,16 @@ myst:
 # Color modes
 
 Bootstrap 5.3 has introduced [Color Modes](https://getbootstrap.com/docs/5.3/customize/color-modes/).
```
```suggestion
Bootstrap 5.3 has introduced [color modes](https://getbootstrap.com/docs/5.3/customize/color-modes/).
```

stevepiercy commented on docs/classic-ui/theming/color-mode.md:
```diff
@@ -141,8 +141,12 @@ See the following example.
+
+## Using color modes in Plone 6.0
+
+There is also the possibility to use the color modes in Plone 6.0 by manually include `barceloneta=3.1.0` and `plone.staticresources=2.1.0`.
```
Manually include... where? `requirements.txt`? Please be explicit.

```suggestion
To use color modes in Plone 6.0, manually include `barceloneta=3.1.0` and `plone.staticresources=2.1.0`, as described in {ref}`pin-the-version-of-an-add-on`.
```

toalba replied:
You right i will add it. Adding the version binding to the `requirements.txt` will do it. If working with mxdev you can add it to the mx.ini, but i think this information is to explicit right? @stevepiercy

stevepiercy replied:
@toalba OK, I think I understand now. I updated my suggestion with a MyST reference to https://6.docs.plone.org/install/manage-add-ons-packages.html#pin-the-version-of-an-add-on. Let's check the build or preview to make sure it renders correctly.

toalba replied:
ah ok, thanks

toalba replied:
The Reference is not showing in the preview, this is unknown terrain for me. can you look? @stevepiercy

stevepiercy reviewed (changes requested):
A couple more corrections that I missed on my first pass, and addition of a `versionadded`.

I also asked a couple of questions, as a couple of things are not clear to me. Thank you!

toalba committed:
Update docs/classic-ui/theming/color-mode.md

adding ref to `pin-the-version-of-an-add-on` and correcting grammar

Co-authored-by: Steve Piercy <web@stevepiercy.com>

toalba committed:
Update docs/classic-ui/theming/color-mode.md

themes --> modes

Co-authored-by: Steve Piercy <web@stevepiercy.com>

toalba committed:
regarding the review changed example id 

`id="exampldescribtioneInputEmail1"` to `id="exampleInputEmail1"`

toalba committed:
Update docs/classic-ui/theming/color-mode.md

Color Modes --> color modes

Co-authored-by: Steve Piercy <web@stevepiercy.com>

stevepiercy commented on docs/classic-ui/theming/color-mode.md:
```diff
@@ -149,4 +149,4 @@ See the following example.
 ## Using color modes in Plone 6.0
 
-There is also the possibility to use the color modes in Plone 6.0 by manually include `barceloneta=3.1.0` and `plone.staticresources=2.1.0`.
+To use color modes in Plone 6.0, manually include `barceloneta=3.1.0` and `plone.staticresources=2.1.0`, as described in {ref}`pin-the-version-of-an-add-on`.
```
My apologies, I gave you the incorrect reference. This time I looked it up in the source.

```suggestion
To use color modes in Plone 6.0, manually include `barceloneta=3.1.0` and `plone.staticresources=2.1.0`, as described in {ref}`manage-add-an-add-on`.
```

stevepiercy committed:
Update docs/classic-ui/theming/color-mode.md

Fix ref

stevepiercy committed:
Merge branch '6-dev' into color-mode-bootstrap5

stevepiercy reviewed (approved):
Let's wait for a green build, and ensure that the preview looks correct https://deploy-preview-1443--6-docs-plone-org.netlify.app/classic-ui/theming/color-mode.html

I'm OK with merging this now. Yes, it's premature, but the momentum is here, and the versioning issue has been addressed adequately, so I'll go with it. We need docs badly.

@petschki @MrTango any last thoughts?

petschki reviewed (approved):
Nice! Thanks @toalba !

toalba commented:
Thanks as well to @lenadax , who initially developed this feature.

rnixx commented:
Then it's time to drop alpha releases of mockup, staticresources and barceloneta including bs 5.3 i guess ;)
